hpx/lcos/future.hpp

```diff
@@ -89,7 +89,9 @@ namespace hpx { namespace lcos { namespace detail
         } else if (state == future_state::invalid) {
             f = Future();
         } else {
-            HPX_ASSERT(false);
+            HPX_THROW_EXCEPTION(invalid_status,
+                "serialize_future_load",
+                "attempting to deserialize a future with an invalid state");
```

Comment:
I think this should read "unknown state" as ´future_state::invalid` seems to be just fine.